ISISScientificComputing/autoreduce
# Cache plots

### Summary of work

Creates a custom django cache backend that is used for storing plots.
The cache extends the filebased cache and stores value with key, value pairs being plot regex string: local plot path
When `is_expired` is called the cache will also delete the plot file locally, the same way the normal cache would delete the key.

I have changed `PlotHandler` to generate the regex pattern inside the `__init__` since it is used in most places and a new `PlotHandler` is created on every access to the `run_summary` view anyway.

Currently the cache expires once per day (Elliots suggestion) but this can be increased or decreased dependant on what we need.

### How to test your work
 
Run a reduction that gets a graph. Reload the page and verify that the plot_cache has been accessed and not ceph.
Make sure plot is deleted on expiry


### Additional comments
 

Fixes #616


**Before merging ensure the release notes have been updated**
